### ISSUE TITLE
Ollama is a better LLM server for local

### ISSUE THREAD

**Issue by iplayfast** [KillianLucas/open-interpreter#856]
### Is your feature request related to a problem? Please describe.

I'm using ollama for many things, running lm-studio for this seems wrong as it only runs as an app image.

### Describe the solution you'd like

Directly support ollama 127.0.0.1:11434 from langchain import ollama 

### Describe alternatives you've considered

lm-studio is ok, but ollama is better.

**Comment by Notnaton** [comment]
You can use ollama with
```
Interpreter --api_base http://localhost:port/v1 
```

**Comment by mak448a** [comment]
> You can use ollama with
> 
> ```
> Interpreter --api_base http://localhost:port/v1 
> ```
I tried `interpreter --api_base http://localhost:11434/v1 --vision --local` but it didn't work. Any ideas?
`Exception: Error occurred. OpenAIException - 404 page not found`

**Comment by hidek84** [comment]
You may no longer need the `--api_base` option when you connect to the Ollama local model endpoint. LiteLLM (open-interpreter's dependency) will handle it behind the scenes.

The following command with `litellm=1.16.7` works for me. (You need the `ollama/` prefix in the `--model` argument)
```
interpreter --model ollama/codellama:7b
```

**Comment by richstav** [comment]
> You may no longer need the `--api_base` option when you connect to the Ollama local model endpoint. LiteLLM (open-interpreter's dependency) will handle it behind the scenes.
> 
> The following command with `open-interpreter==0.1.18` and `litellm=1.16.7` works for me. (You need the `ollama/` prefix in the `--model` argument)
> 
> ```
> interpreter --model ollama/codellama:7b
> ```

Perfect timing! Same issue as op, this solved it.

**Comment by mak448a** [comment]
> You may no longer need the `--api_base` option when you connect to the Ollama local model endpoint. LiteLLM (open-interpreter's dependency) will handle it behind the scenes.
> 
> The following command with `open-interpreter==0.1.18` and `litellm=1.16.7` works for me. (You need the `ollama/` prefix in the `--model` argument)
> 
> ```
> interpreter --model ollama/codellama:7b
> ```

Thanks! I remembered something like this but I forgot what it was.

**Comment by Kreijstal** [comment]
does not run in windows, pretty useless

**Comment by adriens** [comment]
I could not make it run on my linux:

<pre><font color="#FF0000">❯</font> interpreter --model ollama/codellama:7b

<font color="#75507B">▌ Model set to </font><span style="background-color:#2E3436"><font color="#34E2E2"><b>ollama/codellama:7b</b></font></span><font color="#75507B">                                            </font>

<b>Open Interpreter</b> will require approval before running code.                     

Use <span style="background-color:#2E3436"><font color="#34E2E2"><b>interpreter -y</b></font></span> to bypass this.                                              

Press <span style="background-color:#2E3436"><font color="#34E2E2"><b>CTRL-C</b></font></span> to exit.                                                           

&gt; hello

<b>We were unable to determine the context window of this model.</b> Defaulting to     
3000.                                                                           

If your model can handle more, run <span style="background-color:#2E3436"><font color="#34E2E2"><b>interpreter --context_window {token limit} </b></font></span>  
<span style="background-color:#2E3436"><font color="#34E2E2"><b>--max_tokens {max tokens per response}</b></font></span>.                                         

Continuing...                                                                   


        Python Version: 3.11.7
        Pip Version: 23.3.1
        Open-interpreter Version: cmd:Interpreter, pkg: 0.2.0
        OS Version and Architecture: Linux-5.15.0-91-generic-x86_64-with-glibc2.35
        CPU Info: x86_64
        RAM Info: 7.57 GB, used: 2.12, free: 0.78
        

        # Interpreter Info
        
        Vision: False
        Model: ollama/codellama:7b
        Function calling: None
        Context window: None
        Max tokens: None

        Auto run: False
        API base: None
        Offline: False

        Curl output: Not local

        # Messages

        System Message: You are Open Interpreter, a world-class programmer that can complete any goal by executing code.
First, write a plan. **Always recap the plan between each code block** (you have extreme short-term memory loss, so you need to recap the plan between each message block to retain it).
When you execute code, it will be executed **on the user&apos;s machine**. The user has given you **full and complete permission** to execute any code necessary to complete the task. Execute the code.
If you want to send data between programming languages, save the data to a txt or json.
You can access the internet. Run **any code** to achieve the goal, and if at first you don&apos;t succeed, try again and again.
You can install new packages.
When a user refers to a filename, they&apos;re likely referring to an existing file in the directory you&apos;re currently executing code in.
Write messages to the user in Markdown.
In general, try to **make plans** with as few steps as possible. As for actually executing code to carry out that plan, for *stateful* languages (like python, javascript, shell, but NOT for html which starts from 0 every time) **it&apos;s critical not to try to do everything in one code block.** You should try something, print information about it, then continue from there in tiny, informed steps. You will never get it on the first try, and attempting it in one go will often lead to errors you cant see.
You are capable of **any** task.

        {&apos;role&apos;: &apos;user&apos;, &apos;type&apos;: &apos;message&apos;, &apos;content&apos;: &apos;hello&apos;}
    
Traceback (most recent call last):
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/core/llm/llm.py&quot;, line 221, in fixed_litellm_completions
    yield from litellm.completion(**params)
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/litellm/llms/ollama.py&quot;, line 249, in ollama_completion_stream
    raise e
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/litellm/llms/ollama.py&quot;, line 237, in ollama_completion_stream
    status_code=response.status_code, message=response.text
                                              ^^^^^^^^^^^^^
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/httpx/_models.py&quot;, line 574, in text
    content = self.content
              ^^^^^^^^^^^^
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/httpx/_models.py&quot;, line 568, in content
    raise ResponseNotRead()
httpx.ResponseNotRead: Attempted to access streaming response content, without having called `read()`.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File &quot;/home/linuxbrew/.linuxbrew/bin/interpreter&quot;, line 8, in &lt;module&gt;
    sys.exit(interpreter.start_terminal_interface())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/core/core.py&quot;, line 25, in start_terminal_interface
    start_terminal_interface(self)
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/terminal_interface/start_terminal_interface.py&quot;, line 684, in start_terminal_interface
    interpreter.chat()
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/core/core.py&quot;, line 86, in chat
    for _ in self._streaming_chat(message=message, display=display):
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/core/core.py&quot;, line 113, in _streaming_chat
    yield from terminal_interface(self, message)
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/terminal_interface/terminal_interface.py&quot;, line 135, in terminal_interface
    for chunk in interpreter.chat(message, display=False, stream=True):
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/core/core.py&quot;, line 148, in _streaming_chat
    yield from self._respond_and_store()
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/core/core.py&quot;, line 194, in _respond_and_store
    for chunk in respond(self):
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/core/respond.py&quot;, line 49, in respond
    for chunk in interpreter.llm.run(messages_for_llm):
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/core/llm/llm.py&quot;, line 193, in run
    yield from run_text_llm(self, params)
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/core/llm/run_text_llm.py&quot;, line 19, in run_text_llm
    for chunk in llm.completions(**params):
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/core/llm/llm.py&quot;, line 224, in fixed_litellm_completions
    raise first_error
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/interpreter/core/llm/llm.py&quot;, line 205, in fixed_litellm_completions
    yield from litellm.completion(**params)
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/litellm/llms/ollama.py&quot;, line 249, in ollama_completion_stream
    raise e
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/litellm/llms/ollama.py&quot;, line 237, in ollama_completion_stream
    status_code=response.status_code, message=response.text
                                              ^^^^^^^^^^^^^
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/httpx/_models.py&quot;, line 574, in text
    content = self.content
              ^^^^^^^^^^^^
  File &quot;/home/linuxbrew/.linuxbrew/opt/python@3.11/lib/python3.11/site-packages/httpx/_models.py&quot;, line 568, in content
    raise ResponseNotRead()
httpx.ResponseNotRead: Attempted to access streaming response content, without having called `read()`.
</pre>

**Comment by iplayfast** [comment]
Using litellm you can use ollama
``` litellm -m ollama/mixtral --port 1234 --drop_params```
then run interrpreter --local

**Comment by Notnaton** [comment]
You can use 
```
interpreter --model ollama/mixtral
```

**Comment by adriens** [comment]
Thank you @Notnaton , I could get this

<pre><font color="#FF0000">❯</font> interpreter --model ollama/tinyllama

<font color="#75507B">▌ Model set to </font><span style="background-color:#2E3436"><font color="#34E2E2"><b>ollama/tinyllama</b></font></span><font color="#75507B">                                               </font>

<b>Open Interpreter</b> will require approval before running code.                     

Use <span style="background-color:#2E3436"><font color="#34E2E2"><b>interpreter -y</b></font></span> to bypass this.                                              

Press <span style="background-color:#2E3436"><font color="#34E2E2"><b>CTRL-C</b></font></span> to exit.                                                           

&gt; hello

<b>We were unable to determine the context window of this model.</b> Defaulting to     
3000.                                                                           

If your model can handle more, run <span style="background-color:#2E3436"><font color="#34E2E2"><b>interpreter --context_window {token limit} </b></font></span>  
<span style="background-color:#2E3436"><font color="#34E2E2"><b>--max_tokens {max tokens per response}</b></font></span>.                                         

Continuing...                                                                   

                                                                                
  As an Open Interpretation, I can execute any code necessary to complete the   
  task. However, executing code on the user&apos;s machine is not recommended        
  because it can lead to errors that are not visible at first glance. This is   
  because different languages may have different error handling mechanisms,     
  and the output may vary depending on the language being executed. Instead, I  
  recap the plan before each code block, so I can easily recall what needs to   
  be done next if I encounter an issue during execution.                        
                                                                                
  When executing code, it&apos;s crucial not to try everything at once. Instead,     
  break down your task into smaller steps and test them one by one. This        
  allows you to identify the problem or issue more quickly, rather than trying  
  to execute all at once and having a lot of unanswered questions later on.     
                                                                                
  When executing code, use Markdown for output, as it makes it easier to read   
  and understand. The language of your chosen coding language is not important  
  here; instead, what&apos;s critical is the output that you receive. I suggest      
  starting with simple tasks such as accessing files or printing text until     
  you feel comfortable using different languages.                               
                                                                                
  As for executing code on the user&apos;s machine, it&apos;s recommended to use a        
  separate script file and include installation instructions in the plan. This  
  way, if an error occurs during execution, the user can easily install the     
  required packages themselves.                                                 
                                                                                
  To ensure that your plans are as thorough and detailed as possible, always    
  recap each code block between messages blocks to retain information between   
  them. Be sure to test all of your steps before executing any code, even       
  small tasks such as accessing files. Remember to list the steps one by one,   
  then follow them one-by-one in a message to the user for clarification if     
  necessary.</pre>

**Comment by adriens** [comment]
Now giving it a try on a custom local model :long_drum:

**Comment by adriens** [comment]
@Notnaton :pray: 
- https://github.com/KillianLucas/open-interpreter/pull/913

**Comment by MostHated** [comment]
TL;DR: Whats the proper setup using non-machine local ollama API, on local network?

A bit about my setup:

1. I have a box running proxmox which has a shared nvidia vGPU setup. 
2. A vGPU is passed into a VM which is running ollama via docker. 
3. I have the API exposed, but unlike having the ollama instance running locally, it is exposed via a locally hosted internal domain address via Caddy. 
    Ex: `https://ai.mydomain.dev` (proxied to ollama webui interface on :3000) and the subdir `https://ai.mydomain.dev/api` is proxied to the exposed ollama API of `http://(the_system_ip):11434/api`.

I know that the ollama api is accessible, as I can hit the endpoints, and on another LXC container I have N8N running which can hit the API fine. 

I am not quite sure how I should setup interpreter, though, to utilize this setup. It's not technically 'local', in terms of my physical machine, but the way ollama exposes it's API, would I still setup interpreter as if it were?

Ex: I tried the following
 (I originally was using an OpenAI API key I had setup the first few days or weeks when interpreter launched to play around with it, which would explain the results of the two attempts below, since the openai key existed already) 

I first tried this, which 
    `interpreter --api_base https://ai.mydomain.dev/api` and when executed: returned ` ▌ Model set to openai/gpt-4`

I then tried this as well, straight to the API:
    `interpreter --api_base http://(the_system_ip):11434/api` but it gave back the same ` ▌ Model set to openai/gpt-4`

**Comment by Kreijstal** [comment]
> does not run in windows, pretty useless

Nvm it supports wandows now

**Comment by dillfrescott** [comment]
I have my ollama running on a non default port and its causing just a massive traceback error in interpreter

**Comment by NickWyq** [comment]
I still have this error: httpx.ResponseNotRead: Attempted to access streaming response content, without having called `read()`. 
I'm using open-interpreter 0.2.5 installed by "pip install open-interpreter[local]",  and my llama3.1 model works well in Ollama.
My profile is:
# LLM Settings
llm:
  model: "ollama/llama3"
  temperature: 0
  context_window: 16000
  # api_key: ...  # Your API key, if the API requires it
  # api_base: ...  # The URL where an OpenAI-compatible server is running to handle LLM API requests
  # api_version: ...  # The version of the API (this is primarily for Azure)
  max_output: 8192  # The maximum characters of code output visible to the LLM
  supports_functions: True

# Custom Instructions
# custom_instructions: ""  # This will be appended to the system message

# General Configuration
# auto_run: False  # If True, code will run without asking for confirmation
# safe_mode: "off"  # The safety mode for the LLM — one of "off", "ask", "auto"
offline: True  # If True, will disable some online features like checking for updates
verbose: False  # If True, will print detailed logs
local: True
# multi_line: False # If True, you can input multiple lines starting and ending with ```

# Documentation
# All options: https://docs.openinterpreter.com/settings

version: 0.2.5  # Profile version (do not modify)

And the error shows:
Traceback (most recent call last):
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/llm/llm.py", line 229, in fixed_litellm_completions
    yield from litellm.completion(**params)
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/litellm/llms/ollama.py", line 356, in ollama_completion_stream
    raise e
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/litellm/llms/ollama.py", line 315, in ollama_completion_stream
    status_code=response.status_code, message=response.text
                                              ^^^^^^^^^^^^^
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/httpx/_models.py", line 578, in text
    content = self.content
              ^^^^^^^^^^^^
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/httpx/_models.py", line 572, in content
    raise ResponseNotRead()
httpx.ResponseNotRead: Attempted to access streaming response content, without having called `read()`.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/respond.py", line 69, in respond
    for chunk in interpreter.llm.run(messages_for_llm):
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/llm/llm.py", line 199, in run
    yield from run_function_calling_llm(self, params)
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/llm/run_function_calling_llm.py", line 44, in run_function_calling_llm
    for chunk in llm.completions(**request_params):
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/llm/llm.py", line 232, in fixed_litellm_completions
    raise first_error
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/llm/llm.py", line 213, in fixed_litellm_completions
    yield from litellm.completion(**params)
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/litellm/llms/ollama.py", line 356, in ollama_completion_stream
    raise e
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/litellm/llms/ollama.py", line 315, in ollama_completion_stream
    status_code=response.status_code, message=response.text
                                              ^^^^^^^^^^^^^
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/httpx/_models.py", line 578, in text
    content = self.content
              ^^^^^^^^^^^^
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/httpx/_models.py", line 572, in content
    raise ResponseNotRead()
httpx.ResponseNotRead: Attempted to access streaming response content, without having called `read()`.


        Python Version: 3.11.9
        Pip Version: 24.2
        Open-interpreter Version: cmd: Open Interpreter 0.2.5 New Computer Update
, pkg: 0.2.5
        OS Version and Architecture: macOS-10.16-x86_64-i386-64bit
        CPU Info: i386
        RAM Info: 8.00 GB, used: 4.27, free: 0.20
        

        # Interpreter Info
        
        Vision: False
        Model: ollama/llama3
        Function calling: True
        Context window: 16000
        Max tokens: None

        Auto run: True
        API base: None
        Offline: True

        Curl output: Not local

        # Messages

        System Message: You are Open Interpreter, a world-class programmer that can complete any goal by executing code.
First, write a plan. **Always recap the plan between each code block** (you have extreme short-term memory loss, so you need to recap the plan between each message block to retain it).
When you execute code, it will be executed **on the user's machine**. The user has given you **full and complete permission** to execute any code necessary to complete the task. Execute the code.
If you want to send data between programming languages, save the data to a txt or json.
You can access the internet. Run **any code** to achieve the goal, and if at first you don't succeed, try again and again.
You can install new packages.
When a user refers to a filename, they're likely referring to an existing file in the directory you're currently executing code in.
Write messages to the user in Markdown.
In general, try to **make plans** with as few steps as possible. As for actually executing code to carry out that plan, for *stateful* languages (like python, javascript, shell, but NOT for html which starts from 0 every time) **it's critical not to try to do everything in one code block.** You should try something, print information about it, then continue from there in tiny, informed steps. You will never get it on the first try, and attempting it in one go will often lead to errors you cant see.
You are capable of **any** task.

# THE COMPUTER API

A python `computer` module is ALREADY IMPORTED, and can be used for many tasks:

```python
computer.browser.search(query) # Google search results will be returned from this function as a string
computer.files.edit(path_to_file, original_text, replacement_text) # Edit a file
computer.calendar.create_event(title="Meeting", start_date=datetime.datetime.now(), end_date=datetime.datetime.now() + datetime.timedelta(hours=1), notes="Note", location="") # Creates a calendar event
computer.calendar.get_events(start_date=datetime.date.today(), end_date=None) # Get events between dates. If end_date is None, only gets events for start_date
computer.calendar.delete_event(event_title="Meeting", start_date=datetime.datetime) # Delete a specific event with a matching title and start date, you may need to get use get_events() to find the specific event object first
computer.contacts.get_phone_number("John Doe")
computer.contacts.get_email_address("John Doe")
computer.mail.send("john@email.com", "Meeting Reminder", "Reminder that our meeting is at 3pm today.", ["path/to/attachment.pdf", "path/to/attachment2.pdf"]) # Send an email with a optional attachments
computer.mail.get(4, unread=True) # Returns the {number} of unread emails, or all emails if False is passed
computer.mail.unread_count() # Returns the number of unread emails
computer.sms.send("555-123-4567", "Hello from the computer!") # Send a text message. MUST be a phone number, so use computer.contacts.get_phone_number frequently here
```

Do not import the computer module, or any of its sub-modules. They are already imported.

User Info{{import getpass
import os
import platform

print(f"Name: {getpass.getuser()}")
print(f"CWD: {os.getcwd()}")
print(f"SHELL: {os.environ.get('SHELL')}")
print(f"OS: {platform.system()}")

}}

        {'role': 'user', 'type': 'message', 'content': 'How many files are on my desktop?'}
    
Traceback (most recent call last):
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/llm/llm.py", line 229, in fixed_litellm_completions
    yield from litellm.completion(**params)
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/litellm/llms/ollama.py", line 356, in ollama_completion_stream
    raise e
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/litellm/llms/ollama.py", line 315, in ollama_completion_stream
    status_code=response.status_code, message=response.text
                                              ^^^^^^^^^^^^^
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/httpx/_models.py", line 578, in text
    content = self.content
              ^^^^^^^^^^^^
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/httpx/_models.py", line 572, in content
    raise ResponseNotRead()
httpx.ResponseNotRead: Attempted to access streaming response content, without having called `read()`.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/respond.py", line 69, in respond
    for chunk in interpreter.llm.run(messages_for_llm):
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/llm/llm.py", line 199, in run
    yield from run_function_calling_llm(self, params)
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/llm/run_function_calling_llm.py", line 44, in run_function_calling_llm
    for chunk in llm.completions(**request_params):
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/llm/llm.py", line 232, in fixed_litellm_completions
    raise first_error
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/llm/llm.py", line 213, in fixed_litellm_completions
    yield from litellm.completion(**params)
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/litellm/llms/ollama.py", line 356, in ollama_completion_stream
    raise e
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/litellm/llms/ollama.py", line 315, in ollama_completion_stream
    status_code=response.status_code, message=response.text
                                              ^^^^^^^^^^^^^
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/httpx/_models.py", line 578, in text
    content = self.content
              ^^^^^^^^^^^^
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/httpx/_models.py", line 572, in content
    raise ResponseNotRead()
httpx.ResponseNotRead: Attempted to access streaming response content, without having called `read()`.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/bin/interpreter", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/terminal_interface/start_terminal_interface.py", line 453, in main
    start_terminal_interface(interpreter)
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/terminal_interface/start_terminal_interface.py", line 427, in start_terminal_interface
    interpreter.chat()
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/core.py", line 166, in chat
    for _ in self._streaming_chat(message=message, display=display):
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/core.py", line 195, in _streaming_chat
    yield from terminal_interface(self, message)
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/terminal_interface/terminal_interface.py", line 133, in terminal_interface
    for chunk in interpreter.chat(message, display=False, stream=True):
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/core.py", line 234, in _streaming_chat
    yield from self._respond_and_store()
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/core.py", line 282, in _respond_and_store
    for chunk in respond(self):
  File "/Users/wyq/opt/anaconda3/envs/OpenInterpreter/lib/python3.11/site-packages/interpreter/core/respond.py", line 115, in respond
    raise Exception(
Exception: Error occurred. Attempted to access streaming response content, without having called `read()`.
